python/ray/serve/_private/replica.py

```diff
@@ -845,7 +845,9 @@ async def prepare_for_shutdown(self):
         # We set the del method to noop after successfully calling it so the
         # destructor is called only once.
         async with self.delete_lock:
-            self.metrics_pusher = None
+            if self.metrics_pusher:
+                self.metrics_pusher.shutdown()
+                self.metrics_pusher = None
```

Comment:
this is unnecessary now

Reply:
Oh gotcha, since the `shutdown()` on metrics pusher is idempotent, it doesn't matter if we call shutdown multiple times and it's unnecessary to set those to `None` for garbage collect. Will make change to all those!